source/frontend/src/features/mapSideBar/acquisition/tabs/agreement/detail/AgreementView.test.tsx

```diff
@@ -54,8 +57,39 @@ describe('AgreementView component', () => {
     expect(asFragment()).toMatchSnapshot();
   });
 
-  it('renders the agreement type ', () => {
+  it('renders the agreement type', () => {
     const { getByText } = setup();
     expect(getByText(/License Of Occupation/i)).toBeVisible();
   });
+
+  it('calls onGenerate when generation button is clicked', async () => {
+    const { getAllByTitle } = setup();
+
+    const generateButtons = getAllByTitle(/Download File/i);
```

Comment:
nice to have: I'm guessing the title was used here because of the generic "generate" text on the button, could change to the new, specific text of the button instead?

Reply:
correct